spec/fixtures/mauth_signature_testing.json

```diff
@@ -0,0 +1,18 @@
+{
+  "description": "Cross platform testing for mAuth signatures",
+  "private_key": "-----BEGIN RSA PRIVATE KEY-----\nMIIEpAIBAAKCAQEAs3zgudWufUW/UFoQtxfBePRDqc63RdpvuBILKHNFAYpxxuSR\nMzQUKSrsvXpWf983/kTtvXBrj4Hm6qWC9fqGKh/qDTnjamXUzxg6utFbcrOrc4hs\nfvfRjEaymclTPLXRa9+iD3vTzDBqS85eDFDm5hTpPwc3C/7+iw0cfWqwP7jJsn5O\nvul50UOuR+czeBWuQXZPh21nUpEI7ZW8U3W2lmO2DrSez2FVtW3Oiek/AQz+uOtl\nNSPoJcH06O2dlh7AAF8c3eFTaYH4gBTFzcBUp3BneZ2cr0e5m5eVRcCOFqDGQF3Y\nJQT2I1EpMecRWo7UW48mF0wzVj/mK1rQUAebgQIDAQABAoIBACAXpf7UTByuCeUO\nFYsHPlqoIikMgwyEYBFjeIdFBQOfg3Ryjdu/5hLuT+IZK7o1aUeXf4KtxS2lpmoy\nKdZdcvu5NRokTZtKleBpjqa0pEtAANnpfKy/FsKkKW8B5lYmlElbdRibpWUPCxJ+\n1aYSGRbuij3wxlDoyQ6Hy55JIzZhQVaKxFr0YQS/v99EArjlhlMJDNqaFEMQ0WUO\n4h7a2ZwAcJ5aDBpCNqFnWg0Vm4u3GM7l50jC0poQGw7UamDL6DWS5UfontSNfBOe\naQO4998dq2FYZGeJiKbZPLzieb1Dk8CcXXgaXO5Lwyy2Rrux+SaSvMOLNki2e593\nlobKOgECgYEA2GQEymGpVY4cqKFHpPIlU/+PhXG9DcZpNzo1oJOxJ5vruoH3UnwV\nmjrP0/NZsjG+loADAhvdlNzBv5pfzmrdUtN/6+HY7zveWfMpdwUSYUaUq+kXDbFH\nNSHCjz+vS6GuOBZ0LvniMR6sAuV25BFP3oCXUxB0+8/z+iyiP4khqNECgYEA1Fea\ntFiD/4kAYivkEX8S1/628o1uqN6rPTB2t0KFaUvH+Mefqf9akZXWqUPbQc2LZEo9\nD6Ez5o51PRBYkHE/C4gfk6ujkwkl8XTgi5owpY73B75ook5mO1MfRIvfvcP1z2Ta\nR2p+uMUDJVO0C4L+RouUWUpkLP8BPg1rUM0sc7ECgYEAkVG6Fd+4RIiHnoeRAajM\ngLijvc5AVDvm9PvWf9wvoJYJnNsjKPXD3Cua3pASsKTPhWq6mnP0PsByLSaTKKCD\nudfnlJW7hg4CqQ2vzwpM6Z7owPpsTPm9BGWDr4fpRTVzNp99rv6JdMtQYTGQwmEN\n7jMVbOckaOeixWOsIlcJj8ECgYEAmYJ3ylePneZai551dBys79AqTLHoxVas70Ch\nIp2Ju3TYrccLa6e6vzNXC+mNkkXZtvhgqnL9BXoJ0cqGbG4iiOCxC13zlHHxp1y6\nlNI0xwvTFRsXo/cPu2W9Xh3M8/C+PWAI2cZotIVhX9PifswFrdRsvBymzUzRhh3H\nbpPVxhECgYAFRJ3qF9kj3TIqupZRknwvEVYMMOK4L8v8zaYjM9WQobcPkROmu7nn\nBzqqXY1SzEH4xRed4OqtTPtglA4jRC4Kr0V0H7terM9u3+p1OIWl7/5X79DYfqyW\nugSNZFZUII03sibgNFxMnj+5s6Bj3PvKjmUnd+Hrt4upKaHH0aRBtA==\n-----END RSA PRIVATE KEY-----",
+  "attributes_for_signing": {
+    "app_uuid": "5ff4257e-9c16-11e0-b048-0026bbfffe5e",
+    "time": 1309891855,
+    "verb": "PUT",
+    "request_url": "/v1/pictures",
+    "body": "<Use the binary value of the blank.jpeg file for the '~_binary' signatures, and an empty string for the '~_empty' signatures>",
+    "query_string": "key=-_.~ !@#$%^*()+{}|:\"'`<>?&∞=v&キ=v&0=v&a=v&a=b&a=c&a=a&k=&k=v"
```

Comment:
A space is added between `~` and `!`

Reply:
@ykitamura-mdsol : Is not "query_string" need to be encoded  before we pass it to the request? Because `#` sign in the url denotes the last part of url and when we extract query string then we get query string only as `key=-_.~ !@`. 
For e.g. if
 `requestUrl = "https://example.org/v1/pictures?key=-_.~ !@#$%^*()+{}|:\"'`<>?&∞=v&キ=v&0=v&a=v&a=b&a=c&a=a&k=&k=v"`
Then, the query extracted from the `HttpRequestMessage` which have above `requestUrl` as RequestUri), then we get QueryString = `key=-_.~ !@`

https://stackoverflow.com/questions/32564635/not-able-to-read-from-query-string

Reply:
Not needed here. This is a test case to check MAuth protocol implementation itself.

Reply:
In ruby, may be you pass `query_string` separately. What about in python? Do you extract queryString from the RequestUri ? Then in those cases, we  need to EncodeQueryString before creating RequestUri with requestUrlString and later after extracting Query while building `String_to_Sign`, need to decodeQueryString before sending to `BuildEncodedQueryParams` . Am I correct ?

Reply:
https://github.com/mdsol/mauth-client-python/blob/94ed1eb293d16608b9c9a0a2e8ee76c822175584/tests/signer_test.py#L13

Reply:
Seems like does not work in that way in `.Net`, as we extract queryString from RequestMessage. Therefore, while creating requestString need to encodeQueryString and again decodeQueryString later. Else, the "queryString" fetched is truncated with `#` sign.
![image](https://user-images.githubusercontent.com/43785517/70657731-5f390d00-1c2a-11ea-8f1e-0091acd7cbf4.png)

Reply:
If it is related to the `.Net` implementation, we should not keep discussing here. Please open a PR in `mauth-client-dotnet`. 
Also, please make sure not to mix up encoding in MAuth client and in Request.  They are doing very similar encoding, but each should be independent.

Reply:
I was wrong.. 🙇  Basically, we should not care about escaping/unescaping in mauth-client.
Commented here also: https://github.com/mdsol/mauth-client-dotnet/pull/59#discussion_r358837944